### PR TITLE
Tests: added CarTest + authorship comments

### PR DESCRIPTION
**Pull request by rtrembecky**

### CODE REVIEW

File: src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    public void testChangeState() throws Exception {
+        Car car = create();
+
+        car.changeState(CarState.OK);
+        assertEquals(car.getCarState(), CarState.OK);
+
+        car.changeState(CarState.DISCARDED);
+        assertEquals(car.getCarState(), CarState.DISCARDED);
+
+        car.changeState(CarState.SERVICING);
```
Comment (rtrembecky) [inline src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java]: I guess we don't want to cover the case when somebody "missclicks" and discards a car by accident and then wants to fix his mistake, do we? Seems like a possible use case, what do you think @jkuchar ?

File: src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        assertEquals(car.getCarState(), CarState.SERVICING);
+
+        car.changeState(CarState.OK);
+        assertEquals(car.getCarState(), CarState.OK);
+
+        car.changeState(CarState.OK);
+        assertEquals(car.getCarState(), CarState.OK);
+
+        car.changeState(CarState.DISCARDED);
+        assertEquals(car.getCarState(), CarState.DISCARDED);
```
Comment (jkuchar) [inline src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java]: what about checking also fail scenarios?

File: src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        car.setType("top kek");
+        assertEquals(car.getType(), "top kek");
+    }
+
+    @Test
+    public void testSetNumberOfSeats() throws Exception {
+        Car car = create();
+
+        car.setNumberOfSeats(3);
+        assertEquals(car.getNumberOfSeats(), 3);
```
Comment (jkuchar) [inline src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java]: Tests for getter and setter (which does no do any validation) is useless. It is like checking if `a = 5`; assigns five into variable `a`. Please focus more on changeState() which contains business logic. ;-)

Comment (rtrembecky) [inline src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java]: fair point, going to get rid of these

File: src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java [hunk 4]
```diff
@@ -44,46 +48,30 @@ public void testChangeState() throws Exception {
...
-    @Test
-    public void testSetNumberOfSeats() throws Exception {
-        Car car = create();
+        car.changeState(CarState.DISCARDED);
+        assertEquals(car.getCarState(), CarState.DISCARDED);
 
-        car.setNumberOfSeats(3);
-        assertEquals(car.getNumberOfSeats(), 3);
+        try {
+            car.changeState(CarState.OK);
```
Comment (jkuchar) [inline src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java]: try assertThows() ;)

Comment (rtrembecky) [inline src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java]: I am unable to find a good documentation with an example on how to exactly use `assertThrows()`, can you find me any?

**Comment by rtrembecky** [comment]
I guess we can switch to assertThrows later. I would like this to be merged before deadline.

**Comment by jkuchar** [comment]
@rtrembecky makes sense, please create issue for that ;-)

File: src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java (hunk 4 above)
Comment (jkuchar) [inline src/test/java/cz/fi/muni/pa165/Model/Entity/CarTest.java]: https://github.com/jkuchar/PA165-java-project/blob/4eca538a02b099a0f2b9de36c836ecc3d3ff6c5c/src/test/java/cz/fi/muni/pa165/Model/Entity/UserTest.java#L73

**Pull request merged**